### ISSUE TITLE
Implement a wider set of environments such as multiline, etc.

### ISSUE THREAD

**Issue by schneiderfelipe** [tlienart/Franklin.jl#628]
This is something that was commented in #624 (both by @tlienart [here](https://github.com/tlienart/Franklin.jl/issues/624#issuecomment-699514514) and by myself [here](https://github.com/tlienart/Franklin.jl/issues/624#issuecomment-699528958)).

As I see it, the idea would be to to get closer to pure LaTeX when it comes to math, while keeping the limitations of KaTeX (or whatever backend will be supported in the future) in mind. An already working example from [here](https://github.com/tlienart/Franklin.jl/issues/624#issuecomment-699514514):

```
# Some markdown
Etc with _emphasis_ etc

\begin{equation}
\frac{
    \begin{array}[b]{r}
      \left( x_1 x_2 \right)\\
      \times \left( x'_1 x'_2 \right)
    \end{array}
  }{
    \left( y_1y_2y_3y_4 \right)
  }
\end{equation}

Some more markdown. Note that there are no \$ anywhere. 
```

That could be extended to `\begin{gather}`, etc.

**Comment by tlienart** [comment]
Yeah so those would need to be prioritised a bit, adding all of them would be a bit of a pain to maintain (esp with eventual support for two maths back ends)

Which ones do you think are the most useful?

**Comment by schneiderfelipe** [comment]
> Yeah so those would need to be prioritised a bit, adding all of them would be a bit of a pain to maintain (esp with eventual support for two maths back ends)
> 
> Which ones do you think are the most useful?

[`multline`](https://www.overleaf.com/learn/latex/Aligning%20equations%20with%20amsmath#Displaying_long_equations) and [`gather`](https://www.overleaf.com/learn/latex/Aligning%20equations%20with%20amsmath#Grouping_and_centering_equations) are definitely the most useful ones (after the ones already implemented, of course 😃).
